src/Mvc/test/Mvc.FunctionalTests/HtmlGenerationTest.cs

```diff
@@ -137,7 +137,8 @@ public async Task HtmlGenerationWebSite_GeneratesExpectedResults(string action,
             }
         }
 
-        [Fact]
+        [ConditionalFact]
+        [SkipOnHelix("https://github.com/dotnet/aspnetcore/issues/34599")]
```

Comment:
We already do this a few places but it's a **Bad Idea:tm:** all around. The fallout is either the attribute does nothing (if the project sets `<BuildHelixPayload>false</BuildHelixPayload>`) or the test _never_ runs because the project only executes on Helix (the default).

The bug here is in unconditionally using `TestPathUtilities.GetSolutionRootDirectory("Mvc")` _anywhere_ in a Helix test. I don't see how that could be working without addressing the bug nor why the problem emerged in your preview7 PR.

Reply:
I don't either, many of the tests in this class have the same treatment. Hence why I filed the bug and moved on.

We'll have to deal with this in the future by avoiding any reliance on `TestPathUtilities.GetSolutionRootDirectory` in our tests, but that's obviously out of the scope of this PR.

Reply:
> We'll have to deal with this in the future by avoiding any reliance on `TestPathUtilities.GetSolutionRootDirectory` in our tests, but that's obviously out of the scope of this PR.

I agree. In the meantime, don't use `[SkipOnHelix(...)]` without the `Queues` parameter because it isn't clear at all. Just add a `Skip` parameter to the `[Fact]` attribute.

Reply:
Done

Reply:
@dougbu your suggestion didn’t seem to work.

I’m reverting the commit and keeping the original ConditionalFact since that seemed to do the trick.